src/shared/metrics.c

```diff
@@ -0,0 +1,35 @@
+#include "log.h"
+#include "metrics.h"
+#include "mkdir-label.h"
+#include "varlink-io.systemd.Metrics.h"
+#include "varlink-util.h"
+
+int metrics_setup_varlink_server(sd_varlink_server **m, sd_event *event, sd_varlink_method_t vl_method_list_cb, void *userdata, const char *socket) {
+        _cleanup_(sd_varlink_server_unrefp) sd_varlink_server *s = NULL;
+        int r;
+
+        r = varlink_server_new(&s, SD_VARLINK_SERVER_INHERIT_USERDATA, userdata);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate varlink server object: %m");
+
+        r = sd_varlink_server_add_interface(s, &vl_interface_io_systemd_Metrics);
+        if (r < 0)
+                return log_error_errno(r, "Failed to add Metrics interface to varlink server: %m");
+
+        r = sd_varlink_server_bind_method(
```

Comment:
please do not unnecessarily split lines. This one is short enough.

This is what coding style says about it:
```
Don't break code lines too eagerly. We do **not** force line breaks at 80ch,
  all of today's screens should be much larger than that. But then again, don't
  overdo it, ~109ch should be enough really.